includes/integration.php

```diff
@@ -0,0 +1,537 @@
+        {
+            include_once(METRILO_ANALYTICS_PLUGIN_PATH . 'includes/tracking-events.php');
+        }
+        
+        public function add_to_cart(
+            $cart_item_key,
+            $product_id,
+            $quantity,
+            $variation_id = false,
+            $variation = false,
+            $cart_item_data = false
+        )
+        {
+            $product_id = ($variation_id) ? $variation_id : $product_id;
+            $this->put_event_in_cookie_queue("window.metrilo.addToCart('$product_id', '$quantity');"
+            );
+        }
+        
+        public function remove_from_cart($key_id)
+        {
+            if (!is_object($this->woo->cart)) {
+                return true;
+            }
+            $cart_items = $this->woo->cart->get_cart();
+            $removed_cart_item = isset($cart_items[$key_id]) ? $cart_items[$key_id] : false;
+            if ($removed_cart_item) {
+                $product_id = $removed_cart_item['data']->get_id();
+                $quantity = (int)$removed_cart_item['quantity'];
+                $this->put_event_in_cookie_queue("window.metrilo.removeFromCart('$product_id', $quantity);");
+            }
+        }
+        
+        public function new_order_event($order_id) {
+            try {
+                // fetch the order
+                $order                    = new WC_Order($order_id);
+                $serialized_order         = $this->order_serializer->serialize($order);
+                $client                   = $this->api_client->get_client($this->tracking_endpoint_domain);
+                $this->identify_call_data = $order->get_billing_email();
+                
+                $client->order($serialized_order);
+                $this->session_set(
+                    $this->get_do_identify_cookie_name(),
+                    wp_json_encode($this->identify_call_data, JSON_UNESCAPED_UNICODE)
+                );
+            } catch(\Exception $e) {
+                $this->error_logger('new_order_event ', $e->getMessage(), $this->plugin_log_path);
+            }
+        }
+        
+        public function resolve_product($product_id)
+        {
+            if (function_exists('wc_get_product')) {
+                return wc_get_product($product_id);
+            }
+        }
+        
+        public function put_event_in_queue($event)
+        {
+            if ($this->check_if_event_should_be_ignored($event)) {
+                return true;
+            }
+            
+            array_push($this->events_queue, $event);
+        }
+        
+        public function put_event_in_cookie_queue($event)
+        {
+            if ($this->check_if_event_should_be_ignored($event)) {
+                return true;
+            }
+            
+            $this->add_item_to_cookie($event);
+        }
+        
+        public function check_if_event_should_be_ignored($event)
+        {
+            if (empty($this->ignore_for_events)) {
```

Comment:
You can set `ignore_for_events` to `[]` when initializing it and then in this method will be something like

```php
public function check_if_event_should_be_ignored($event)
{
  return in_array($event, $this->ignore_for_events);
}
```

Reply:
Fixed.